### ISSUE TITLE
Mouse behavior enhancements

### ISSUE THREAD

**Issue by bobbylight** [bobbylight/RSyntaxTextArea#59]
Originally from issue #56:
- Double-click-hold selects a word, but dragging should then select at word boundaries, not character boundaries.
- Triple-click-hold doesn't select the row, so I can't use it to select multiple rows.

**Comment by peternlewis** [comment]
Back in 2011 I added support for double-click-and-drag and triple-click-and-drag to Processing (see the JEditTextArea.java changes in commit: https://github.com/processing/processing/commit/cec8209ff67058e0938ca19707db053c1e90093e).  I did it to get the fixes in to Arduino, but apparently since then they changed to RSyntaxTextArea.  It would be nice to get these sort of changes in to RSyntaxTextArea and hence into Arduino.  I'm not sure how applicable my patch would be, but it might be helpful.  I took at look at the RSyntaxTextArea code, but with all the features in this I have no idea where to even start.  But if I can help, I would be happy to do so.
